pandas/tests/scalar/interval/test_arithmetic.py

```diff
@@ -0,0 +1,54 @@
+from datetime import timedelta
+
+import numpy as np
+import pytest
+
+from pandas import Interval, Timedelta, Timestamp
+
+
+@pytest.mark.parametrize("method", ["__add__", "__sub__"])
+@pytest.mark.parametrize(
+    "delta", [Timedelta(days=7), timedelta(7), np.timedelta64(7, "D")]
+)
+def test_timestamp_interval_add_subtract_timedelta(method, delta):
+    # https://github.com/pandas-dev/pandas/issues/32023
+    interval = Interval(
+        Timestamp("2017-01-01 00:00:00"), Timestamp("2018-01-01 00:00:00")
+    )
+
+    result = getattr(interval, method)(delta)
+
+    left = getattr(interval.left, method)(delta)
+    right = getattr(interval.right, method)(delta)
+    expected = Interval(left, right)
+
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "delta", [Timedelta(days=7), timedelta(7), np.timedelta64(7, "D")]
+)
+def test_timedelta_add_timestamp_interval(delta):
+    # https://github.com/pandas-dev/pandas/issues/32023
+    interval = Interval(
+        Timestamp("2017-01-01 00:00:00"), Timestamp("2018-01-01 00:00:00")
+    )
```

Comment:
should we get a Timedelta interval here too?  also some NaTs

Reply:
Is it possible to put NaT in an Interval?

Reply:
Intervals can't contain NA values as endpoints.  It doesn't look like this gracefully handles adding `NaT` to intervals with `Timestamp`/`Timedelta` endpoints, but this is also true for numeric intervals, so I'm fine fixing the general case of interval arithmetic with NA in a separate PR.
```python
In [1]: import numpy as np; import pandas as pd; pd.__version__
Out[1]: '1.1.0.dev0+1299.g6e492722d'

In [2]: pd.Interval(0, 1) + np.nan
---------------------------------------------------------------------------
ValueError: left side of interval must be <= right side

In [3]: pd.Interval(pd.Timestamp("2020"), pd.Timestamp("2021")) + pd.NaT
---------------------------------------------------------------------------
TypeError: unsupported operand type(s) for +: 'pandas._libs.interval.Interval' and 'NaTType'
```